src/process.c

```diff
@@ -560,9 +953,22 @@ cc_oci_vm_launch (struct cc_oci_config *config)
 	}
 
 out:
-	g_strfreev (args);
+	if (net_cfg_pipe[0] != -1) close (net_cfg_pipe[0]);
+	if (net_cfg_pipe[1] != -1) close (net_cfg_pipe[1]);
+	if (child_err_pipe[0] != -1) close (child_err_pipe[0]);
+	if (child_err_pipe[1] != -1) close (child_err_pipe[1]);
+
+	/* TODO: Free the network configuration buffers */
+
+	if (args) {
+		g_strfreev (args);
+	}
 
 	return ret;
+
+child_failed:
+	(void)write (child_err_pipe[1], "E", 1);
```

Comment:
It doesn't have to be "E", but if the child returns any data (of length 1 byte), that is considered an error, hence I chose "`E`" to denote error :-). It could probably do with a comment to explain that above this line.